service/src/test/java/aero/minova/cas/service/FilesServiceTest.java

```diff
@@ -0,0 +1,124 @@
+import aero.minova.cas.api.domain.Table;
+import aero.minova.cas.api.domain.Value;
+import aero.minova.cas.controller.BaseTest;
+import aero.minova.cas.controller.SqlViewController;
+
+//benötigt, damit JUnit-Tests nicht abbrechen
+@SpringBootTest(properties = { "application.runner.enabled=false" })
+public class FilesServiceTest extends BaseTest {
+
+	@Autowired
+	FilesService testSubject;
+
+	@Mock
+	SqlViewController mockController;
+
+	@DisplayName("MDI Test mit Masken und Actions")
+	@WithMockUser(username = "user", roles = {})
+	@Test
+	public void testMdi() throws Exception {
+
+		Table mockResult = new Table();
+		mockResult.addColumn(new Column("ID", DataType.STRING));
+		mockResult.addColumn(new Column("Icon", DataType.STRING));
+		mockResult.addColumn(new Column("Label", DataType.STRING));
+		mockResult.addColumn(new Column("Menu", DataType.STRING));
+		mockResult.addColumn(new Column("Position", DataType.DOUBLE));
+		mockResult.addColumn(new Column("SecurityToken", DataType.STRING));
+		mockResult.addColumn(new Column("MdiTypeKey", DataType.INTEGER));
+		{
+			Row inputRow = new Row();
+			inputRow.addValue(new Value("cussom97286TrafficMeasurement", null));
+			inputRow.addValue(new Value("TrafficMeasurement", null));
+			inputRow.addValue(new Value("@xtcussom97286TrafficMeasurement", null));
+			inputRow.addValue(new Value("cussom97286Tables", null));
+			inputRow.addValue(new Value(2.0, null));
+			inputRow.addValue(null);
+			inputRow.addValue(new Value(1, null));
+			mockResult.addRow(inputRow);
+		}
+		{
+			Row inputRow = new Row();
+			inputRow.addValue(new Value("cussom97286Voucher", null));
+			inputRow.addValue(new Value("Voucher", null));
+			inputRow.addValue(new Value("@xtcussom97286Voucher", null));
+			inputRow.addValue(new Value("cussom97286Tables", null));
+			inputRow.addValue(new Value(1.0, null));
+			inputRow.addValue(null);
+			inputRow.addValue(new Value(1, null));
+			mockResult.addRow(inputRow);
+		}
+		{
+			Row inputRow = new Row();
+			inputRow.addValue(new Value("cussom97286Tables", null));
+			inputRow.addValue(null);
+			inputRow.addValue(new Value("@cussom97286.mdi.Tables", null));
+			inputRow.addValue(null);
+			inputRow.addValue(new Value(1.0, null));
+			inputRow.addValue(null);
+			inputRow.addValue(new Value(2, null));
+			mockResult.addRow(inputRow);
+		}
+		{
+			Row inputRow = new Row();
+			inputRow.addValue(null);
+			inputRow.addValue(new Value("SIS", null));
+			inputRow.addValue(new Value("@cussom97286.Title", null));
+			inputRow.addValue(null);
+			inputRow.addValue(new Value(1.0, null));
+			inputRow.addValue(null);
+			inputRow.addValue(new Value(3, null));
+			mockResult.addRow(inputRow);
+		}
+
+		doReturn(mockResult).when(mockController).getIndexView(Mockito.any());
+
+		testSubject.setViewController(mockController);
+
+		byte[] awatingResult = new byte[] { 60, 63, 120, 109, 108, 32, 118, 101, 114, 115, 105, 111, 110, 61, 34, 49, 46, 48, 34, 32, 101, 110, 99, 111, 100,
```

Comment:
Bitte das Ergebniss in eine Datei schreiben und diese beim Test auslesen und vergleichen.